networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/jvmlibp2p/Peer.java

```diff
@@ -63,7 +64,7 @@ public boolean isConnected() {
   }
 
   public CompletableFuture<StatusData> sendStatus() {
-    return send(RpcMethod.STATUS, statusMessageFactory.createStatusMessage())
+    return requestSingle(RpcMethod.STATUS, statusMessageFactory.createStatusMessage())
```

Comment:
One idea to simplify the API for callers would be to make `requestSingle`, `requestStream`, and `send` private methods, and then publicly expose methods for specific requests like `requestBlocks(roots)`.

Reply:
Agreed.